com.hazelcast.jet.cdc.postgres.MultiTableCacheIntegrationTest.ordersOfCustomers [HZ-1741]

_master_ (commit 4252b4f340beedefaea4aec193d60e93a2ea9c28)

Failed on ZuluJDK17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZuluJDK17-nightly/32/testReport/com.hazelcast.jet.cdc.postgres/MultiTableCacheIntegrationTest/ordersOfCustomers/

It can be related https://github.com/hazelcast/hazelcast/issues/21746

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<{1001=Customer {id=1001, firstName=Sally, lastName=Thomas, email=sally.thomas@acme.com}, Orders: {10001=Order {orderNumber=10001, orderDate=Sat Jan 16 00:00:00 UTC 2016, purchaser=1001, quantity=1, productId=102}}, 1002=Customer {id=1002, firstName=George, lastName=Bailey, email=gbailey@foobar.com}, Orders: {10002=Order {orderNumber=10002, orderDate=Sun Jan 17 00:00:00 UTC 2016, purchaser=1002, quantity=2, productId=105}, 10007=Order {orderNumber=10007, orderDate=Fri Feb 19 00:00:00 UTC 2016, purchaser=1002, quantity=2, productId=106}}, 1003=Customer {id=1003, firstName=Edward, lastName=Walker, email=ed@walker.com}, Orders: {10004=Order {orderNumber=10004, orderDate=Sun Feb 21 00:00:00 UTC 2016, purchaser=1003, quantity=1000, productId=107}}, 1004=Customer {id=1004, firstName=Anne1000, lastName=Kretchmar, email=annek@noanswer.org}, Orders: {}, 1005=null, Orders: {}}> but was:<{1001=Customer {id=1001, firstName=Sally, lastName=Thomas, email=sally.thomas@acme.com}, Orders: {10001=Order {orderNumber=10001, orderDate=Sat Jan 16 00:00:00 UTC 2016, purchaser=1001, quantity=1, productId=102}}, 1002=Customer {id=1002, firstName=George, lastName=Bailey, email=gbailey@foobar.com}, Orders: {10002=Order {orderNumber=10002, orderDate=Sun Jan 17 00:00:00 UTC 2016, purchaser=1002, quantity=2, productId=105}, 10007=Order {orderNumber=10007, orderDate=Fri Feb 19 00:00:00 UTC 2016, purchaser=1002, quantity=2, productId=106}}, 1003=Customer {id=1003, firstName=Edward, lastName=Walker, email=ed@walker.com}, Orders: {10004=Order {orderNumber=10004, orderDate=Sun Feb 21 00:00:00 UTC 2016, purchaser=1003, quantity=1000, productId=107}}, 1004=Customer {id=1004, firstName=Anne1, lastName=Kretchmar, email=annek@noanswer.org}, Orders: {}, 1005=null, Orders: {}}>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at com.hazelcast.test.HazelcastTestSupport.lambda$assertEqualsEventually$8(HazelcastTestSupport.java:1005)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1236)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1338)
	at com.hazelcast.test.HazelcastTestSupport.assertEqualsEventually(HazelcastTestSupport.java:1005)
	at com.hazelcast.jet.cdc.postgres.MultiTableCacheIntegrationTest.ordersOfCustomers(MultiTableCacheIntegrationTest.java:126)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

</details>

<details><summary>Standard output:</summary>

```
07:19:17,251 DEBUG |ordersOfCustomers| - [PrefixingImageNameSubstitutor] Time-limited test - No prefix is configured
07:19:17,251 DEBUG |ordersOfCustomers| - [ImageNameSubstitutor] Time-limited test - Did not find a substitute image for debezium/example-postgres:1.7 (using image substitutor: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor'))
07:19:17,251 DEBUG |ordersOfCustomers| - [AbstractImagePullPolicy] Time-limited test - Using locally available and not pulling image: debezium/example-postgres:1.7
07:19:17,252  INFO |ordersOfCustomers| - [7]] Time-limited test - Creating container for image: debezium/example-postgres:1.7
07:19:17,252 DEBUG |ordersOfCustomers| - [RegistryAuthLocator] Time-limited test - Looking up auth config for image: debezium/example-postgres:1.7 at registry: https://index.docker.io/v1/
07:19:17,252 DEBUG |ordersOfCustomers| - [RegistryAuthLocator] Time-limited test - No matching Auth Configs - falling back to defaultAuthConfig [null]
07:19:17,252 DEBUG |ordersOfCustomers| - [AuthDelegatingDockerClientConfig] Time-limited test - Effective auth config [null]
07:19:17,273  INFO |ordersOfCustomers| - [7]] Time-limited test - Container debezium/example-postgres:1.7 is starting: 369e590beeb026710c8b289db44aa5653749c8dc1c9497bb6469264f252033e2
07:19:17,694 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: fixing permissions on existing directory /var/lib/postgresql/data ... ok
07:19:17,694 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:19:17,694 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: Data page checksums are disabled.
07:19:17,694 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:19:17,694 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: The default text search configuration will be set to "english".
07:19:17,694 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: The default database encoding has accordingly been set to "UTF8".
07:19:17,694 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: The database cluster will be initialized with locale "en_US.utf8".
07:19:17,694 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:19:17,694 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: This user must also own the server process.
07:19:17,694 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: The files belonging to this database system will be owned by user "postgres".
07:19:17,705 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: creating subdirectories ... ok
07:19:17,715 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: selecting default max_connections ... 100
07:19:17,737 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: selecting default shared_buffers ... 128MB
07:19:17,759 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: selecting dynamic shared memory implementation ... posix
07:19:17,759 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: selecting default timezone ... Etc/UTC
07:19:17,770 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: creating configuration files ... ok
07:19:17,897 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: running bootstrap script ... ok
07:19:18,333 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: performing post-bootstrap initialization ... ok
07:19:18,454 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:19:18,454 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT:     pg_ctl -D /var/lib/postgresql/data -l logfile start
07:19:18,454 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:19:18,454 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: Success. You can now start the database server using:
07:19:18,454 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:19:18,454 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: syncing data to disk ... ok
07:19:18,454 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: --auth-local and --auth-host, the next time you run initdb.
07:19:18,454 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: You can change this by editing pg_hba.conf or using the option -A, or
07:19:18,454 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: WARNING: enabling "trust" authentication for local connections
07:19:18,454 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: 
07:19:18,493 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: waiting for server to start....2022-09-09 07:19:18.492 GMT [48] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
07:19:18,503 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 2022-09-09 07:19:18.499 GMT [49] LOG:  database system was shut down at 2022-09-09 07:19:18 GMT
07:19:18,513 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 2022-09-09 07:19:18.504 GMT [48] LOG:  database system is ready to accept connections
07:19:18,581 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: server started
07:19:18,581 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT:  done
07:19:18,647 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:19:18,648 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: /usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/inventory.sql
07:19:18,648 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: /usr/local/bin/docker-entrypoint.sh: sourcing /docker-entrypoint-initdb.d/init-permissions.sh
07:19:18,648 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:19:18,698 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: SET
07:19:18,698 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE SCHEMA
07:19:19,456 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE EXTENSION
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER TABLE
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE TABLE
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 9
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER TABLE
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER SEQUENCE
07:19:19,467 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE TABLE
07:19:19,477 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 2022-09-09 07:19:19.476 GMT [75] FATAL:  terminating autovacuum process due to administrator command
07:19:19,477 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 2022-09-09 07:19:19.476 GMT [48] LOG:  aborting any active transactions
07:19:19,477 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: waiting for server to shut down....2022-09-09 07:19:19.476 GMT [48] LOG:  received fast shutdown request
07:19:19,477 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:19:19,477 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:19:19,477 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 3
07:19:19,477 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE TABLE
07:19:19,477 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 4
07:19:19,477 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER TABLE
07:19:19,477 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER SEQUENCE
07:19:19,477 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE TABLE
07:19:19,477 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 4
07:19:19,477 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER TABLE
07:19:19,478 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER SEQUENCE
07:19:19,478 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE TABLE
07:19:19,488 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 2022-09-09 07:19:19.479 GMT [48] LOG:  background worker "logical replication launcher" (PID 55) exited with exit code 1
07:19:19,498 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 2022-09-09 07:19:19.490 GMT [50] LOG:  shutting down
07:19:19,508 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 2022-09-09 07:19:19.506 GMT [48] LOG:  database system is shut down
07:19:19,577 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:19:19,577 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: PostgreSQL init process complete; ready for start up.
07:19:19,577 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:19:19,577 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: server stopped
07:19:19,577 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT:  done
07:19:19,596 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: 2022-09-09 07:19:19.596 GMT [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
07:19:19,596 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: 2022-09-09 07:19:19.595 GMT [1] LOG:  listening on IPv6 address "::", port 5432
07:19:19,596 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: 2022-09-09 07:19:19.595 GMT [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
07:19:19,609 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: 2022-09-09 07:19:19.608 GMT [77] LOG:  database system was shut down at 2022-09-09 07:19:19 GMT
07:19:19,619 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: 2022-09-09 07:19:19.613 GMT [1] LOG:  database system is ready to accept connections
07:19:19,620  INFO |ordersOfCustomers| - [7]] Time-limited test - Container debezium/example-postgres:1.7 started in PT2.368601022S
07:19:19,627  INFO |ordersOfCustomers| - [MetricsConfigHelper] ordersOfCustomers - [LOCAL] [dev] [5.2-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
07:19:19,628  INFO |ordersOfCustomers| - [logo] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
07:19:19,628  INFO |ordersOfCustomers| - [system] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
07:19:19,628  INFO |ordersOfCustomers| - [system] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Hazelcast Platform 5.2-SNAPSHOT (20220909 - 4252b4f) starting at [127.0.0.1]:5701
07:19:19,628  INFO |ordersOfCustomers| - [system] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Cluster name: dev
07:19:19,628  INFO |ordersOfCustomers| - [system] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
07:19:19,628  INFO |ordersOfCustomers| - [system] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Jet is enabled
07:19:19,630  INFO |ordersOfCustomers| - [MetricsConfigHelper] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
07:19:19,634  WARN |ordersOfCustomers| - [CPSubsystem] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
07:19:19,637  INFO |ordersOfCustomers| - [JetServiceBackend] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
07:19:19,638  INFO |ordersOfCustomers| - [Diagnostics] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
07:19:19,638  INFO |ordersOfCustomers| - [LifecycleService] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is STARTING
07:19:19,639  INFO |ordersOfCustomers| - [ClusterService] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - 62d1e6c8-bbd9-4dbb-aaca-505b0f332f11 this
]

07:19:19,639  INFO |ordersOfCustomers| - [JobCoordinationService] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Jet started scanning for jobs
07:19:19,639  INFO |ordersOfCustomers| - [LifecycleService] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is STARTED
07:19:19,641  INFO |ordersOfCustomers| - [PartitionStateManager] hz.upbeat_shaw.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initializing cluster partition table arrangement...
07:19:19,648  INFO |ordersOfCustomers| - [JobCoordinationService] hz.upbeat_shaw.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Starting job 089c-865d-d5c0-0001 based on submit request
07:19:19,650  INFO |ordersOfCustomers| - [MasterJobContext] hz.upbeat_shaw.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Didn't find any snapshot to restore for job '089c-865d-d5c0-0001', execution 089c-865d-d5c1-0001
07:19:19,650  INFO |ordersOfCustomers| - [MasterJobContext] hz.upbeat_shaw.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Start executing job '089c-865d-d5c0-0001', execution 089c-865d-d5c1-0001, execution graph in DOT format:
digraph DAG {
	"source" [localParallelism=1];
	"filter-2" [localParallelism=2];
	"filter" [localParallelism=2];
	"mapWithEntryProcessorSink(cache)-2" [localParallelism=2];
	"mapWithEntryProcessorSink(cache)" [localParallelism=2];
	"source" -> "filter" [taillabel=0, queueSize=1024];
	"source" -> "filter-2" [taillabel=1, queueSize=1024];
	"filter-2" -> "mapWithEntryProcessorSink(cache)-2" [label="partitioned", queueSize=1024];
	"filter" -> "mapWithEntryProcessorSink(cache)" [label="partitioned", queueSize=1024];
}
HINT: You can use graphviz or http://viz-js.com/ to visualize the printed graph.
07:19:19,653  INFO |ordersOfCustomers| - [JobExecutionService] hz.upbeat_shaw.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Execution plan for jobId=089c-865d-d5c0-0001, jobName='089c-865d-d5c0-0001', executionId=089c-865d-d5c1-0001 initialized
07:19:19,653  INFO |ordersOfCustomers| - [JobExecutionService] hz.upbeat_shaw.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Start execution of job '089c-865d-d5c0-0001', execution 089c-865d-d5c1-0001 from coordinator [127.0.0.1]:5701
07:19:19,655  INFO |ordersOfCustomers| - [ChangeRecordCdcSourceP] hz.upbeat_shaw.jet.blocking.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [089c-865d-d5c0-0001/source#0] source has 'retry strategy' set to 'max attempts = -1, interval function = IntervalFunction (constant of 1000ms)'
07:19:19,655  INFO |ordersOfCustomers| - [ChangeRecordCdcSourceP] hz.upbeat_shaw.jet.blocking.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [089c-865d-d5c0-0001/source#0] source has 'commit period' set to '10000000000'
07:19:19,655  INFO |ordersOfCustomers| - [ChangeRecordCdcSourceP] hz.upbeat_shaw.jet.blocking.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [089c-865d-d5c0-0001/source#0] source has 'clear state on reconnect' set to 'false'
07:19:19,655  INFO |ordersOfCustomers| - [ChangeRecordCdcSourceP] hz.upbeat_shaw.jet.blocking.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [089c-865d-d5c0-0001/source#0] starting new task with config: {reconnect.behavior=max attempts = -1, interval function = IntervalFunction (constant of 1000ms), connector.class=io.debezium.connector.postgresql.PostgresConnector, database.dbname=postgres, database.user=postgres, db.specific.extra.fields=schema, database.server.name=4601e118-4209-412f-91c3-dc591ffea835, sequence.extractor.class=com.hazelcast.jet.cdc.postgres.impl.PostgresSequenceExtractor, database.port=53259, table.whitelist=inventory.customers,inventory.orders, tombstones.on.delete=false, database.hostname=172.17.0.1, database.password=postgres, name=source, database.history=com.hazelcast.jet.cdc.impl.CdcSourceP$DatabaseHistoryImpl, snapshot.mode=initial}
07:19:19,655  WARN |ordersOfCustomers| - [PostgresConnectorConfig] hz.upbeat_shaw.jet.blocking.thread-0 - Configuration property 'truncate.handling.mode' is deprecated and will be removed in future versions. Please use 'skipped.operations' instead.
07:19:19,656  WARN |ordersOfCustomers| - [Configuration] hz.upbeat_shaw.jet.blocking.thread-0 - Using configuration property "table.whitelist" is deprecated and will be removed in future versions. Please use "table.include.list" instead.
07:19:19,656  WARN |ordersOfCustomers| - [PostgresConnectorConfig] hz.upbeat_shaw.jet.blocking.thread-0 - Configuration property 'toasted.value.placeholder' is deprecated and will be removed in future versions. Please use 'unavailable.value.placeholder' instead.
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 - Starting PostgresConnectorTask with configuration:
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    reconnect.behavior = max attempts = -1, interval function = IntervalFunction (constant of 1000ms)
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    connector.class = io.debezium.connector.postgresql.PostgresConnector
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    database.dbname = postgres
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    database.user = postgres
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    db.specific.extra.fields = schema
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    database.server.name = 4601e118-4209-412f-91c3-dc591ffea835
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    sequence.extractor.class = com.hazelcast.jet.cdc.postgres.impl.PostgresSequenceExtractor
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    database.port = 53259
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    table.whitelist = inventory.customers,inventory.orders
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    tombstones.on.delete = false
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    database.hostname = 172.17.0.1
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    database.password = ********
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    name = source
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    database.history = com.hazelcast.jet.cdc.impl.CdcSourceP$DatabaseHistoryImpl
07:19:19,656  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 -    snapshot.mode = initial
07:19:19,665  INFO |ordersOfCustomers| - [JdbcConnection] pool-190-thread-1 - Connection gracefully closed
07:19:19,684  INFO |ordersOfCustomers| - [BaseSourceTask] hz.upbeat_shaw.jet.blocking.thread-0 - No previous offsets found
07:19:19,686  INFO |ordersOfCustomers| - [PostgresConnectorTask] hz.upbeat_shaw.jet.blocking.thread-0 - user 'postgres' connected to database 'postgres' on PostgreSQL 11.14 (Debian 11.14-1.pgdg90+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516, 64-bit with roles:
	role 'pg_read_all_settings' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_stat_scan_tables' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_write_server_files' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_monitor' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_read_server_files' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_execute_server_program' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_read_all_stats' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_signal_backend' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'postgres' [superuser: true, replication: true, inherit: true, create role: true, create db: true, can log in: true]
07:19:19,687  INFO |ordersOfCustomers| - [PostgresConnection] hz.upbeat_shaw.jet.blocking.thread-0 - Obtained valid replication slot ReplicationSlot [active=false, latestFlushedLsn=null, catalogXmin=null]
07:19:19,687  INFO |ordersOfCustomers| - [PostgresConnectorTask] hz.upbeat_shaw.jet.blocking.thread-0 - No previous offset found
07:19:19,687  INFO |ordersOfCustomers| - [InitialSnapshotter] hz.upbeat_shaw.jet.blocking.thread-0 - Taking initial snapshot for new datasource
07:19:19,692  INFO |ordersOfCustomers| - [PostgresReplicationConnection] hz.upbeat_shaw.jet.blocking.thread-0 - Creating replication slot with command CREATE_REPLICATION_SLOT "debezium"  LOGICAL decoderbufs
07:19:19,694  INFO |ordersOfCustomers| - [Threads] hz.upbeat_shaw.jet.blocking.thread-0 - Requested thread factory for connector PostgresConnector, id = 4601e118-4209-412f-91c3-dc591ffea835 named = change-event-source-coordinator
07:19:19,694  INFO |ordersOfCustomers| - [Threads] hz.upbeat_shaw.jet.blocking.thread-0 - Creating thread debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator
07:19:19,695  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Metrics registered
07:19:19,695  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Context created
07:19:19,695  INFO |ordersOfCustomers| - [InitialSnapshotter] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Taking initial snapshot for new datasource
07:19:19,695  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - According to the connector configuration data will be snapshotted
07:19:19,695  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Snapshot step 1 - Preparing
07:19:19,695  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Setting isolation level
07:19:19,695  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Opening transaction with statement SET TRANSACTION ISOLATION LEVEL REPEATABLE READ; 
SET TRANSACTION SNAPSHOT '00000004-00000002-1';
07:19:19,711  WARN |ordersOfCustomers| - [SchemaNameAdjuster] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - The Kafka Connect schema name '4601e118-4209-412f-91c3-dc591ffea835.inventory.customers.Value' is not a valid Avro schema name, so replacing with '_601e118_4209_412f_91c3_dc591ffea835.inventory.customers.Value'
07:19:19,711  WARN |ordersOfCustomers| - [SchemaNameAdjuster] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - The Kafka Connect schema name '4601e118-4209-412f-91c3-dc591ffea835.inventory.customers.Key' is not a valid Avro schema name, so replacing with '_601e118_4209_412f_91c3_dc591ffea835.inventory.customers.Key'
07:19:19,711  WARN |ordersOfCustomers| - [SchemaNameAdjuster] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - The Kafka Connect schema name '4601e118-4209-412f-91c3-dc591ffea835.inventory.customers.Envelope' is not a valid Avro schema name, so replacing with '_601e118_4209_412f_91c3_dc591ffea835.inventory.customers.Envelope'
07:19:19,711  WARN |ordersOfCustomers| - [SchemaNameAdjuster] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - The Kafka Connect schema name '4601e118-4209-412f-91c3-dc591ffea835.inventory.orders.Value' is not a valid Avro schema name, so replacing with '_601e118_4209_412f_91c3_dc591ffea835.inventory.orders.Value'
07:19:19,711  WARN |ordersOfCustomers| - [SchemaNameAdjuster] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - The Kafka Connect schema name '4601e118-4209-412f-91c3-dc591ffea835.inventory.orders.Key' is not a valid Avro schema name, so replacing with '_601e118_4209_412f_91c3_dc591ffea835.inventory.orders.Key'
07:19:19,711  WARN |ordersOfCustomers| - [SchemaNameAdjuster] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - The Kafka Connect schema name '4601e118-4209-412f-91c3-dc591ffea835.inventory.orders.Envelope' is not a valid Avro schema name, so replacing with '_601e118_4209_412f_91c3_dc591ffea835.inventory.orders.Envelope'
07:19:19,711  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Snapshot step 2 - Determining captured tables
07:19:19,712  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Adding table inventory.geom to the list of capture schema tables
07:19:19,712  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Adding table inventory.products_on_hand to the list of capture schema tables
07:19:19,712  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Adding table inventory.customers to the list of capture schema tables
07:19:19,712  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Adding table inventory.orders to the list of capture schema tables
07:19:19,712  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Adding table inventory.products to the list of capture schema tables
07:19:19,713  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Snapshot step 3 - Locking captured tables [inventory.customers, inventory.orders]
07:19:19,713  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Snapshot step 4 - Determining snapshot offset
07:19:19,713  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Creating initial offset context
07:19:19,713  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Read xlogStart at 'LSN{0/2038BC0}' from transaction '596'
07:19:19,713  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Read xlogStart at 'LSN{0/2038BC0}' from transaction '596'
07:19:19,713  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Snapshot step 5 - Reading structure of captured tables
07:19:19,713  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Reading structure of schema 'inventory' of catalog 'postgres'
07:19:19,734  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Snapshot step 6 - Persisting schema history
07:19:19,734  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Snapshot step 7 - Snapshotting data
07:19:19,734  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Snapshotting contents of 2 tables while still in transaction
07:19:19,734  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Exporting data from table 'inventory.customers' (1 of 2 tables)
07:19:19,734  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - 	 For table 'inventory.customers' using select statement: 'SELECT "id", "first_name", "last_name", "email" FROM "inventory"."customers"'
07:19:19,737  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - 	 Finished exporting 4 records for table 'inventory.customers'; total duration '00:00:00.003'
07:19:19,737  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Exporting data from table 'inventory.orders' (2 of 2 tables)
07:19:19,737  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - 	 For table 'inventory.orders' using select statement: 'SELECT "id", "order_date", "purchaser", "quantity", "product_id" FROM "inventory"."orders"'
07:19:19,740  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - 	 Finished exporting 4 records for table 'inventory.orders'; total duration '00:00:00.003'
07:19:19,741  INFO |ordersOfCustomers| - [AbstractSnapshotChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Snapshot - Final stage
07:19:19,741  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Snapshot ended with SnapshotResult [status=COMPLETED, offset=PostgresOffsetContext [sourceInfoSchema=Schema{io.debezium.connector.postgresql.Source:STRUCT}, sourceInfo=source_info[server='4601e118-4209-412f-91c3-dc591ffea835'db='postgres', lsn=LSN{0/2038BC0}, txId=596, timestamp=2022-09-09T07:19:19.740Z, snapshot=FALSE, schema=inventory, table=orders], lastSnapshotRecord=true, lastCompletelyProcessedLsn=null, lastCommitLsn=null, streamingStoppingLsn=null, transactionContext=TransactionContext [currentTransactionId=null, perTableEventCount={}, totalEventCount=0], incrementalSnapshotContext=IncrementalSnapshotContext [windowOpened=false, chunkEndPosition=null, dataCollectionsToSnapshot=[], lastEventKeySent=null, maximumKey=null]]]
07:19:19,741  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Connected metrics set to 'true'
07:19:19,752  INFO |ordersOfCustomers| - [PostgresSchema] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - REPLICA IDENTITY for 'inventory.customers' is 'FULL'; UPDATE AND DELETE events will contain the previous values of all the columns
07:19:19,753  INFO |ordersOfCustomers| - [PostgresSchema] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - REPLICA IDENTITY for 'inventory.orders' is 'FULL'; UPDATE AND DELETE events will contain the previous values of all the columns
07:19:19,753  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Starting streaming
07:19:19,753  INFO |ordersOfCustomers| - [PostgresStreamingChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Retrieved latest position from stored offset 'LSN{0/2038BC0}'
07:19:19,753  INFO |ordersOfCustomers| - [WalPositionLocator] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Looking for WAL restart position for last commit LSN 'null' and last change LSN 'LSN{0/2038BC0}'
07:19:19,761  INFO |ordersOfCustomers| - [PostgresConnection] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Obtained valid replication slot ReplicationSlot [active=false, latestFlushedLsn=LSN{0/2038BC0}, catalogXmin=596]
07:19:19,761  INFO |ordersOfCustomers| - [JdbcConnection] pool-191-thread-1 - Connection gracefully closed
07:19:19,772  INFO |ordersOfCustomers| - [Threads] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Requested thread factory for connector PostgresConnector, id = 4601e118-4209-412f-91c3-dc591ffea835 named = keep-alive
07:19:19,773  INFO |ordersOfCustomers| - [Threads] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Creating thread debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-keep-alive
07:19:19,786  INFO |ordersOfCustomers| - [PostgresSchema] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - REPLICA IDENTITY for 'inventory.customers' is 'FULL'; UPDATE AND DELETE events will contain the previous values of all the columns
07:19:19,787  INFO |ordersOfCustomers| - [PostgresSchema] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - REPLICA IDENTITY for 'inventory.orders' is 'FULL'; UPDATE AND DELETE events will contain the previous values of all the columns
07:19:19,787  INFO |ordersOfCustomers| - [PostgresStreamingChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Searching for WAL resume position
07:19:20,799  INFO |ordersOfCustomers| - [WalPositionLocator] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - First LSN 'LSN{0/203CA30}' received
07:19:20,799  INFO |ordersOfCustomers| - [PostgresStreamingChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - WAL resume position 'LSN{0/203CA30}' discovered
07:19:20,800  INFO |ordersOfCustomers| - [JdbcConnection] pool-192-thread-1 - Connection gracefully closed
07:19:20,817  INFO |ordersOfCustomers| - [Threads] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Requested thread factory for connector PostgresConnector, id = 4601e118-4209-412f-91c3-dc591ffea835 named = keep-alive
07:19:20,817  INFO |ordersOfCustomers| - [Threads] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Creating thread debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-keep-alive
07:19:20,817  INFO |ordersOfCustomers| - [PostgresStreamingChangeEventSource] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Processing messages
07:19:20,817  INFO |ordersOfCustomers| - [WalPositionLocator] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Message with LSN 'LSN{0/203CA30}' arrived, switching off the filtering
07:21:21,436  INFO || - [BaseSourceTask] ForkJoinPool.commonPool-worker-7 - Stopping down connector
07:21:21,936  INFO |ordersOfCustomers| - [JdbcConnection] pool-193-thread-1 - Connection gracefully closed
07:21:21,936  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Finished streaming
07:21:21,936  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-4601e118-4209-412f-91c3-dc591ffea835-change-event-source-coordinator - Connected metrics set to 'false'
07:21:21,937  INFO || - [JdbcConnection] pool-194-thread-1 - Connection gracefully closed
07:21:21,938 ERROR |ordersOfCustomers| - [StartExecutionOperation] hz.upbeat_shaw.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] CANCEL_FORCEFUL
com.hazelcast.jet.impl.exception.JobTerminateRequestedException: CANCEL_FORCEFUL
	at com.hazelcast.jet.impl.execution.ExecutionContext.terminateExecution(ExecutionContext.java:318) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution0(JobExecutionService.java:719) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution(JobExecutionService.java:715) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.operation.TerminateExecutionOperation.doRun(TerminateExecutionOperation.java:59) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.operation.AsyncOperation.run(AsyncOperation.java:54) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:281) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:256) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:217) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:411) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:438) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:601) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:580) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:541) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:241) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:61) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipant(MasterContext.java:294) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipants(MasterContext.java:277) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$cancelExecutionInvocations$20(MasterJobContext.java:643) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
07:21:21,941  INFO |ordersOfCustomers| - [MasterJobContext] hz.upbeat_shaw.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Execution of job '089c-865d-d5c0-0001', execution 089c-865d-d5c1-0001 got terminated, reason=java.util.concurrent.CancellationException
	Start time: 2022-09-09T07:19:19.648
	Duration: 00:02:02.292
	To see additional job metrics enable JobConfig.storeMetricsAfterJobCompletion
07:21:21,942 ERROR |ordersOfCustomers| - [JoinSubmittedJobOperation] hz.upbeat_shaw.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] null
java.util.concurrent.CancellationException: null
	at com.hazelcast.jet.impl.MasterJobContext.getErrorFromResponses(MasterJobContext.java:547) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$invokeStartExecution$9(MasterJobContext.java:500) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterContext.lambda$invokeOnParticipant$3(MasterContext.java:312) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.util.ExceptionUtil.lambda$withTryCatch$0(ExceptionUtil.java:175) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.AbstractInvocationFuture$WhenCompleteNode.lambda$execute$0(AbstractInvocationFuture.java:1564) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.test.TestLoggingUtils$CustomTestNameAwareForkJoinPool$TestNameAwareRunnable.run(TestLoggingUtils.java:104) ~[hazelcast-5.2-SNAPSHOT-tests.jar:5.2-SNAPSHOT]
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395) ~[?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]
07:21:21,943 ERROR |ordersOfCustomers| - [JoinSubmittedJobOperation] hz.upbeat_shaw.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] null
java.util.concurrent.CancellationException: null
	at com.hazelcast.jet.impl.MasterJobContext.getErrorFromResponses(MasterJobContext.java:547) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$invokeStartExecution$9(MasterJobContext.java:500) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterContext.lambda$invokeOnParticipant$3(MasterContext.java:312) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.util.ExceptionUtil.lambda$withTryCatch$0(ExceptionUtil.java:175) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.AbstractInvocationFuture$WhenCompleteNode.lambda$execute$0(AbstractInvocationFuture.java:1564) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.test.TestLoggingUtils$CustomTestNameAwareForkJoinPool$TestNameAwareRunnable.run(TestLoggingUtils.java:104) ~[hazelcast-5.2-SNAPSHOT-tests.jar:5.2-SNAPSHOT]
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395) ~[?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]
07:21:21,943  INFO |ordersOfCustomers| - [JetTestSupport] Time-limited test - Terminating instanceFactory in JetTestSupport.@After
07:21:21,944  INFO |ordersOfCustomers| - [LifecycleService] Thread-60 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
07:21:21,944  WARN |ordersOfCustomers| - [Node] Thread-60 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Terminating forcefully...
07:21:21,945  INFO |ordersOfCustomers| - [Node] Thread-60 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Shutting down connection manager...
07:21:21,945  INFO |ordersOfCustomers| - [Node] Thread-60 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Shutting down node engine...
07:21:21,950  INFO |ordersOfCustomers| - [NodeExtension] Thread-60 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Destroying node NodeExtension.
07:21:21,950  INFO |ordersOfCustomers| - [Node] Thread-60 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Hazelcast Shutdown is completed in 6 ms.
07:21:21,950  INFO |ordersOfCustomers| - [LifecycleService] Thread-60 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
07:21:22,319 DEBUG |ordersOfCustomers| - [ResourceReaper] Time-limited test - Removed container and associated volume(s): debezium/example-postgres:1.7
BuildInfo right after ordersOfCustomers(com.hazelcast.jet.cdc.postgres.MultiTableCacheIntegrationTest): BuildInfo{version='5.2-SNAPSHOT', build='20220909', buildNumber=20220909, revision=4252b4f, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'ordersOfCustomers(com.hazelcast.jet.cdc.postgres.MultiTableCacheIntegrationTest):'
07:19:15, accumulated pauses: 67 ms, max pause: 3 ms, pauses over 1000 ms: 0
07:19:20, accumulated pauses: 68 ms, max pause: 1 ms, pauses over 1000 ms: 0
07:19:25, accumulated pauses: 67 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:19:30, accumulated pauses: 72 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:19:35, accumulated pauses: 65 ms, max pause: 1 ms, pauses over 1000 ms: 0
07:19:40, accumulated pauses: 62 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:19:45, accumulated pauses: 70 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:19:50, accumulated pauses: 66 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:19:55, accumulated pauses: 55 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:20:00, accumulated pauses: 49 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:20:05, accumulated pauses: 57 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:20:10, accumulated pauses: 76 ms, max pause: 6 ms, pauses over 1000 ms: 0
07:20:15, accumulated pauses: 70 ms, max pause: 3 ms, pauses over 1000 ms: 0
07:20:20, accumulated pauses: 45 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:20:25, accumulated pauses: 49 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:20:30, accumulated pauses: 54 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:20:35, accumulated pauses: 63 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:20:40, accumulated pauses: 64 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:20:45, accumulated pauses: 66 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:20:50, accumulated pauses: 65 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:20:55, accumulated pauses: 70 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:21:00, accumulated pauses: 66 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:21:05, accumulated pauses: 68 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:21:10, accumulated pauses: 66 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:21:15, accumulated pauses: 74 ms, max pause: 7 ms, pauses over 1000 ms: 0
07:21:20, accumulated pauses: 396 ms, max pause: 370 ms, pauses over 1000 ms: 0
```

</details>

--- a/extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java
+++ b/extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java
@@ -19,6 +19,7 @@ package com.hazelcast.jet.cdc.postgres;
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.jet.cdc.ChangeRecord;
 import com.hazelcast.jet.cdc.ParsingException;
+import com.hazelcast.jet.config.JobConfig;
 import com.hazelcast.jet.pipeline.Pipeline;
 import com.hazelcast.jet.pipeline.Sinks;
 import com.hazelcast.jet.pipeline.StreamSource;
@@ -44,6 +45,7 @@ import java.util.stream.Collectors;
 import static com.hazelcast.jet.cdc.Operation.DELETE;
 import static com.hazelcast.jet.cdc.Operation.INSERT;
 import static com.hazelcast.jet.cdc.Operation.SYNC;
+import static com.hazelcast.jet.config.ProcessingGuarantee.EXACTLY_ONCE;
 import static com.hazelcast.jet.impl.util.ExceptionUtil.rethrow;
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
@@ -52,7 +54,7 @@ import static java.util.Objects.requireNonNull;
 public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrationTest {
 
     private static final String CACHE = "cache";
-    private static final int REPEATS = 1000;
+    private static final int REPEATS = 1001;
 
     @Test
     public void ordersOfCustomers() throws Exception {
@@ -61,8 +63,10 @@ public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrati
                 .build();
 
         Pipeline pipeline = Pipeline.create();
+        pipeline.setPreserveOrder(true);
         StreamStage<ChangeRecord> allRecords = pipeline.readFrom(source)
-                .withNativeTimestamps(0).setLocalParallelism(1);
+                                                       .withoutTimestamps()
+                                                       .setLocalParallelism(1);
 
         allRecords.filter(r -> r.table().equals("customers"))
                 .writeTo(Sinks.mapWithEntryProcessor(CACHE,
@@ -77,8 +81,9 @@ public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrati
                 ));
 
         // when
-        HazelcastInstance hz = createHazelcastInstances(1)[0];
-        hz.getJet().newJob(pipeline);
+        HazelcastInstance hz = createHazelcastInstance();
+        hz.getJet().newJob(pipeline, new JobConfig().setProcessingGuarantee(EXACTLY_ONCE));
+
         //then
         Map<Integer, OrdersOfCustomer> expected = toMap(
         new OrdersOfCustomer(
@@ -96,15 +101,15 @@ public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrati
         assertEqualsEventually(() -> getIMapContent(hz), expected);
 
         //when
+        executeBatch("INSERT INTO customers VALUES (1005, 'Jason', 'Bourne', 'jason@bourne.org')");
         List<String> batch = new ArrayList<>();
-        batch.add("INSERT INTO customers VALUES (1005, 'Jason', 'Bourne', 'jason@bourne.org')");
         for (int i = 1; i <= REPEATS; i++) {
-            batch.add("UPDATE customers SET first_name='Anne" + i + "' WHERE id=1004");
-            batch.add("UPDATE orders SET quantity='" + i + "' WHERE id=10004");
-
-            batch.add("DELETE FROM orders WHERE id=10003");
+            batch.addAll(createTestSqlQueries(i));
+            if (batch.size() >= 50 || i == REPEATS) {
+                executeBatch(batch.toArray(new String[0]));
+                batch.clear();
+            }
         }
-        executeBatch(batch.toArray(new String[0]));
         executeBatch("DELETE FROM customers WHERE id=1005");
         executeBatch("INSERT INTO orders VALUES (10007, '2016-02-19', 1002, 2, 106)");
 
@@ -124,6 +129,15 @@ public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrati
                         new Customer(1004, "Anne" + REPEATS, "Kretchmar", "annek@noanswer.org")));
         expected.put(1005, new OrdersOfCustomer());
         assertEqualsEventually(() -> getIMapContent(hz), expected);
+    }
+
+    private static List<String> createTestSqlQueries(int index) {
+        List<String> batch = new ArrayList<>();
+        batch.add("UPDATE customers SET first_name='Anne" + index + "' WHERE id=1004");
+        batch.add("UPDATE orders SET quantity='" + index + "' WHERE id=10004");
+
+        batch.add("DELETE FROM orders WHERE id=10003");
+        return batch;
     }
 
     @Nonnull

